mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -49,48 +50,46 @@
 
 public class CassandraAnnotationMapper extends NonTransactionalMapper implements AnnotationMapper {
 
-    private final CassandraId mailboxId;
     private final Session session;
 
-    public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
-        this.mailboxId = mailboxId;
+    public CassandraAnnotationMapper(Session session) {
         this.session = session;
     }
 
-    public List<MailboxAnnotation> getAllAnnotations() {
-        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQuery()))
+    public List<MailboxAnnotation> getAllAnnotations(MailboxId mailboxId) {
+        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQuery((CassandraId)mailboxId)))
             .map(this::toAnnotation)
             .collect(Collectors.toList());
     }
 
-    public List<MailboxAnnotation> getAnnotationsByKeys(Set<MailboxAnnotationKey> keys) {
-        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQueryForKeys(keys)))
+    public List<MailboxAnnotation> getAnnotationsByKeys(MailboxId mailboxId, Set<MailboxAnnotationKey> keys) {
+        return CassandraUtils.convertToStream(session.execute(getStoredAnnotationsQueryForKeys((CassandraId)mailboxId, keys)))
             .map(this::toAnnotation)
             .collect(Collectors.toList());
     }
 
-    public List<MailboxAnnotation> getAnnotationsByKeysWithOneDepth(Set<MailboxAnnotationKey> keys) {
+    public List<MailboxAnnotation> getAnnotationsByKeysWithOneDepth(MailboxId mailboxId, Set<MailboxAnnotationKey> keys) {
         return keys.stream()
-            .flatMap(this::getAnnotationsByKeyWithOneDepth)
+            .flatMap(annotation -> getAnnotationsByKeyWithOneDepth((CassandraId) mailboxId, annotation))
```

Comment:
You are casting `mailboxId` on each element of the stream, you may do it outside of the lambda.